theano/tensor/tests/test_basic.py

```diff
@@ -1683,7 +1718,7 @@ def test_py_c_match():
     expected_gamma = scipy.special.gamma
     expected_gammaln = scipy.special.gammaln
     expected_psi = scipy.special.psi
-    expected_chi2sf = lambda x, df: scipy.stats.chi2.sf(x, df).astype(x.dtype)
+    expected_chi2sf = lambda x, df: scipy.stats.chi2.sf(x, df)
```

Comment:
Add this back. This is what make travis fail. With that small changes we should be able to merge this PR.

The problem is that we don't use the same output dtype as scipy for this. They upcast the output to float64, but we force the output to be the same as the input. This is why we needed that small change.

thanks